app/views/unattended/provisioning_templates/provision/autoyast_default.erb

```diff
@@ -223,12 +243,72 @@ rm /etc/resolv.conf
         </source>
       </script>
     </chroot-scripts>
+<% if puppet_enabled -%>
+    <pre-scripts config:type="list">
+      <script>
+        <filename>puppet-keys.sh</filename>
+        <interpreter>shell</interpreter>
+        <notification>Retrieving Puppet repository keys ...</notification>
+        <source><![CDATA[
+<%
+  # Pull Puppetlabs RPM keys manually, automatic Zypper handling with key IDs does not work
+  puppet_repo_url_base = 'http://yum.puppet.com'
+
+  %w[puppet puppetlabs puppet-20250406].each { |key|
+-%>
+rpmkeys --import http://yum.puppet.com/RPM-GPG-KEY-<%= key %>
+<% } -%>
+]]>
+        </source>
+      </script>
+    </pre-scripts>
+<% end -%>
   </scripts>
+<% if os_major >= 15 -%>
+  <firewall>
+    <enable_firewall config:type="boolean">true</enable_firewall>
+    <log_denied_packets>off</log_denied_packets>
+    <default_zone>public</default_zone>
+    <zones config:type="list">
+      <zone>
+        <name>public</name>
+        <services config:type="list">
+          <service>ssh</service>
+          <service>dhcpv6-client</service>
+        </services>
+      </zone>
+    </zones>
+  </firewall>
+<% end -%>
   <keyboard>
     <keymap>english-us</keymap>
   </keyboard>
   <timezone>
     <hwclock>UTC</hwclock>
     <timezone><%= host_param('time-zone') || 'Etc/UTC' %></timezone>
   </timezone>
+  <add-on>
+    <add_on_products config:type="list">
+<% if puppet_enabled && aio_enabled -%>
+<%
+  puppet_repo_url_base = 'http://yum.puppet.com'
+  if host_param_true?('enable-official-puppet7-repo')
+    puppet_repo_url = "#{puppet_repo_url_base}/puppet7/sles/#{os_major}/#{@host.architecture}/"
+  elsif host_param_true?('enable-puppetlabs-puppet6-repo')
+    puppet_repo_url = "#{puppet_repo_url_base}/puppet6/sles/#{os_major}/#{@host.architecture}/"
+  elsif host_param_true?('enable-puppetlabs-puppet5-repo')
+    puppet_repo_url = "#{puppet_repo_url_base}/puppet5/sles/#{os_major}/#{@host.architecture}/"
```

Comment:
I don't think adding Puppet 6 and 5 in 2024 is a good idea :)

Reply:
Puppet 6 is still available in the yum repo at least, but dropped both - and added Puppet 8